### PR TITLE
Fix code coverage.

### LINKED ISSUE #5265: grpc_coverage is not generating reports
@nicolasnoble @jtattermusch any ideas?

### PR DESCRIPTION
**Pull request by adelez**
Fixes #5265.
Ran locally 
tools/run_tests/run_tests.py --use_docker --measure_cpu_costs -t -l all -c gcov -x report.xml -j 2

The report looks good.

### CODE REVIEW

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -850,12 +850,15 @@ def git(cmd, cwd=cwd):
     print 'copied to the docker environment.'
     time.sleep(5)
 
-  dockerfile_dirs = set([l.dockerfile_dir() for l in languages])
-  if len(dockerfile_dirs) > 1:
-    print 'Languages to be tested require running under different docker images.'
-    sys.exit(1)
-  dockerfile_dir = next(iter(dockerfile_dirs))
-
+  if 'gcov' in args.config:
```
Comment (jtattermusch) [inline tools/run_tests/run_tests.py]: we should only use the multilang image when absolutely necessary
e.g.   -c gcov -l csharp  should still run under csharp image.

So i'd say let's only use the multilang image for 'gcov' and len(dockerfile_dirs) > 1.
Perhaps we could also print a message that the multilang image is getting used.

Comment (adelez) [inline tools/run_tests/run_tests.py]: Done. PTAL.

**Comment by jtattermusch** [comment]
LGTM, but we can only merge once the fixit is over.

**Comment by adelez** [comment]
This PR is a fixit itself, not a new feature.

**Comment by jtattermusch** [comment]
Please rebase and push, I can't merge this with tests being red.

**Comment by adelez** [comment]
Done

**Pull request merged**